rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -3257,6 +3251,24 @@ private boolean isXmlWorkflow(InputPart fileInputPart) {
                 .contains(MediaType.APPLICATION_XML.toLowerCase());
     }
 
+    protected static Map<String, String> createSortableTaskAttrMap() {
+        HashMap<String, String> sortableTaskAttrMap = new HashMap<>(13);
+        sortableTaskAttrMap.put("id", "id.taskId");
+        sortableTaskAttrMap.put("status", "taskStatus");
+        sortableTaskAttrMap.put("name", "taskName");
+        sortableTaskAttrMap.put("tag", "tag");
+        sortableTaskAttrMap.put("execDuration", "executionDuration");
+        sortableTaskAttrMap.put("nodeCount", "parallelEnvNodesNumber");
+        sortableTaskAttrMap.put("executions", "numberOfExecutionLeft");
+        sortableTaskAttrMap.put("nodeFailure", "numberOfExecutionOnFailureLeft");
+        sortableTaskAttrMap.put("host", "executionHostName");
+        sortableTaskAttrMap.put("startTime", "startTime");
+        sortableTaskAttrMap.put("finishedTime", "finishedTime");
+        sortableTaskAttrMap.put("description", "description");
+        sortableTaskAttrMap.put("scheduledAt", "scheduledTime");
+        return sortableTaskAttrMap;
+    }
+
     @Override
     public RestPage<String> getTaskIds(String sessionId, long from, long to, boolean mytasks, boolean running,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)